### ISSUE TITLE
ApplicationTests::ServerTest fails locally: "cannot load such file -- rails/commands"

### ISSUE THREAD

**Issue by utilum** [rails/rails#33240]
I don't see this on the CI, but locally it never fails to fail.

```
$  ruby -w -Itest -Ilib -I../activesupport/lib -I../actionpack/lib -I../actionview/lib -I../activemodel/lib test/application/server_test.rb -n="/#test_restart_rails_server_with_custom_pid_file_path/" --backtrace
Run options: -n=/#test_restart_rails_server_with_custom_pid_file_path/ --backtrace --seed 16146

# Running:

F

Failure:
ApplicationTests::ServerTest#test_restart_rails_server_with_custom_pid_file_path [test/application/server_test.rb:53]:
"Inherited" expected, but got:

...
Traceback (most recent call last):
        2: from bin/rails:4:in `<main>'
        1: from /home/user/.rbenv/versions/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
/home/user/.rbenv/versions/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require': cannot load such file -- rails/commands (LoadError)
.
Expected "...\r\n\e[1mTraceback\e[m (most recent call last):\r\n\t2: from bin/rails:4:in `<main>'\r\n\t1: from /home/user/.rbenv/versions/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'\r\n/home/user/.rbenv/versions/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require': \e[1mcannot load such file -- rails/commands (\e[1;4mLoadError\e[m\e[1m)\r\n\e[m" to include "Inherited".


bin/rails test test/application/server_test.rb:35



Finished in 13.620866s, 0.0734 runs/s, 0.4405 assertions/s.
1 runs, 6 assertions, 1 failures, 0 errors, 0 skips
```

**Comment by deivid-rodriguez** [comment]
I can confirm this issue. Run into it when running tests locally for #33202.

**Comment by utilum** [comment]
I run this on Fedora, seeing it with Ruby 2.4, 2.5, and 2.6-dev.

**Comment by utilum** [comment]
This looks like another aspect of #31193. Test passes with bundler 1.15.

**Comment by deivid-rodriguez** [comment]
But Travis seems to be using latest bundler, right? Which ruby version manager (if any) are you using? Me, rbenv.

**Comment by utilum** [comment]
Me too, `rbenv`.

> 1: from /home/user/.rbenv/versions/2.5.1/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'

**Comment by deivid-rodriguez** [comment]
Oooops, didn't look at your backtrace :sweat_smile:

**Comment by matthewd** [comment]
Me three. :confused:

But I get a different error inside the failure:
```
Traceback (most recent call last):
	18: from bin/rails:4:in `<main>'
	17: from /Users/matthew/.rbenv/versions/2.5.0/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	16: from /Users/matthew/.rbenv/versions/2.5.0/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
	15: from /Users/matthew/.rbenv/versions/2.5.0/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require'
	14: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/commands.rb:16:in `<top (required)>'
	13: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/command.rb:44:in `invoke'
	12: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/command/base.rb:63:in `perform'
	11: from /Users/matthew/.rbenv/gems/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	10: from /Users/matthew/.rbenv/gems/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	 9: from /Users/matthew/.rbenv/gems/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	 8: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/commands/server/server_command.rb:130:in `perform'
	 7: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/commands/server/server_command.rb:130:in `tap'
	 6: from /Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/commands/server/server_command.rb:133:in `block in perform'
	 5: from /Users/matthew/.rbenv/versions/2.5.0/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:122:in `require'
	 4: from /Users/matthew/.rbenv/versions/2.5.0/lib/ruby/site_ruby/2.5.0/rubygems/core_ext/kernel_require.rb:122:in `require'
	 3: from /private/var/folders/6s/9rhzk53d5t59lthnn2l_mvhr0000gn/T/d20180628-16048-1x4gwmy/app/config/application.rb:9:in `<top (required)>'
	 2: from /private/var/folders/6s/9rhzk53d5t59lthnn2l_mvhr0000gn/T/d20180628-16048-1x4gwmy/app/config/application.rb:10:in `<module:AppTemplate>'
	 1: from /private/var/folders/6s/9rhzk53d5t59lthnn2l_mvhr0000gn/T/d20180628-16048-1x4gwmy/app/config/application.rb:12:in `<class:Application>'
/Users/matthew/.rbenv/gems/2.5.0/gems/railties-5.1.5/lib/rails/application/configuration.rb:81:in `load_defaults': Unknown version "6.0" (RuntimeError)
```

Assuming that's not some other, unrelated, misbehaviour on my end, it sounds like we're in the vicinity of 91a4a820feeb878dd1b388befa56dd469933d17c: it should _not_ be trying to use my installed 5.1.5 instance.

In common between the two errors, is it suspicious that there's no mention of bundler in the `require` backtrace?

**Comment by utilum** [comment]
I'd uninstalled `rails` and `railties` to get the shorter message. After `gem install rails` I get the same failure as you, @matthewd .

**Comment by utilum** [comment]
This issue still reproduces on master.

**Comment by utilum** [comment]
@rails-bot, this is still active.

**Comment by stevensjared4** [comment]
Issue still occurs when using `rbenv` with `ruby 2.6.3p62 (2019-04-16 revision 67580 [x86_64-linux]`
Going through steps to create a `rails` application listed [here](https://www.digitalocean.com/community/tutorials/how-to-install-ruby-on-rails-with-rbenv-on-ubuntu-18-04). Installing `bundler 2.0.2` and installing `rails 5.2.3` works fine initially. 
The `~/.bashrc` file is updated accordingly and things go smoothly until computer is turned off or reset.
Have to reinstall `bundler 2.0.2, rake 12.3.3, rails 5.2.3` and all dependencies all over again. 
Any ideas as to a fix in `Ubuntu 19.04`?
